tskv/src/tsm/reader.rs

```diff
@@ -303,22 +327,42 @@ pub async fn load_index_v2(tsm_id: u64, reader: Arc<AsyncFile>) -> ReadTsmResult
 
     // Decode index data
     let assumed_field_count = (data_len / (INDEX_META_SIZE + BLOCK_META_SIZE)) + 1;
-    let mut field_id_offs: Vec<(FieldId, usize)> = Vec::with_capacity(assumed_field_count);
+    let mut field_id_idx_offs: Vec<(FieldId, usize)> = Vec::with_capacity(assumed_field_count);
     let mut pos = 0_usize;
+    let mut field_id_of_blocks_is_ordered = true;
+    let mut previous_field_id = 0_u64;
+    let mut previous_field_block_off = 0_u64;
     while pos < data_len {
-        field_id_offs.push((decode_be_u64(&data[pos..pos + 8]), pos));
+        let field_id = decode_be_u64(&data[pos..pos + 8]);
```

Comment:
tsm version mark  for sorted

Reply:
Add enum `TsmVersion` for this condition, if `TsmVersion` >= `V2` then `TsmCache` is enabled.